lsp/nls/src/linearization/mod.rs

```diff
@@ -51,6 +51,11 @@ impl LinRegistry {
         let lin = self.map.get(&id.file_id).unwrap();
         lin.get_item(id)
     }
+
+    /// Remove a file's linearization from the registry.
+    pub fn remove_file(&mut self, file_id: FileId) {
```

Comment:
I don't mind adding such helpers, but it must be noted that in many many places, we are simply using directly `lin_registry.map.xxx`. In particular `insert`. It looks a bit inconsistent to add a `remove_file` but not an `insert_file`, for example. I'm fine either way (using directly `registry.map.xxx` or adding common methods): adding such methods achieve better encapsulation, but it's also noisy boilerplate because we simply wrap `self.map.xxx`, which isn't too bad either. But at least I think we should be consistent.

Reply:
Sounds good, I've removed it for now.